### ISSUE TITLE
Node.js v10.11.0 support

### ISSUE THREAD

**Issue by JTonda** [zowe/zowe-cli-db2-plugin#5]
_From @tucker01 on September 25, 2018 23:18_

I believe this is a known issue, going to documenting here...

On a system with Node.js v10.11.0, building the project from source results in the following error:
```
make: *** [Release/obj.target/odbc_bindings/src/odbc.o] Error 1
gyp ERR! build error
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/home/jason/.nvm/versions/node/v10.11.0/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:262:23)
gyp ERR! stack     at ChildProcess.emit (events.js:182:13)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:240:12)
gyp ERR! System Linux 4.15.0-34-generic
gyp ERR! command "/home/jason/.nvm/versions/node/v10.11.0/bin/node" "/home/jason/.nvm/versions/node/v10.11.0/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "configure" "build" "--IS_DOWNLOADED=true" "--IBM_DB_HOME=/home/jason/projects/zowe-cli-db2-plugin/node_modules/ibm_db/installer/clidriver"
gyp ERR! cwd /home/jason/projects/zowe-cli-db2-plugin/node_modules/ibm_db
gyp ERR! node -v v10.11.0
gyp ERR! node-gyp -v v3.8.0
gyp ERR! not ok

    at ChildProcess.exithandler (child_process.js:289:12)
    at ChildProcess.emit (events.js:182:13)
    at maybeClose (internal/child_process.js:962:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:251:5)
  killed: false,
  code: 1,
  signal: null,
  cmd:
   'node-gyp configure build  --IS_DOWNLOADED=true --IBM_DB_HOME="$IBM_DB_HOME"' }
```

Attempting to install via `plugins install` command results in the following error:

``
prebuild-install WARN install No prebuilt binaries found (target=10.11.0 runtime=node arch=x64 platform=linux)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! ibm_db@2.4.0 install: `node installer/driverInstall.js`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the ibm_db@2.4.0 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/jason/.npm/_logs/2018-09-25T23_13_19_056Z-debug.log
Command Error:
Install Failed
Error Details:
Command failed: npm install "@brightside/db2@next" --prefix /home/jason/.brightside/plugins/installed -g --registry https://registry.npmjs.org/
``

Both tests were performed from an x64 Ubuntu 18.04.1 LTS and from a Docker image using `FROM ubuntu:xenial`. 

For reference, the [DB2](https://github.com/ibmdb/node-ibm_db) repo does not list Node.js v10 as a compatible release. 

_Copied from original issue: gizafoundation/zowe-cli-db2-plugin#3_

**Comment by JTonda** [comment]
_From @AHumanFromCA on September 26, 2018 14:17_

I also opened this issue a while ago somewhere. Copying it here for future reference:

The DB2 plugin is unusable on Brightside when running under Node 10.

Validate Output:

```
$ bright plugins validate @brightside/db2

Validation results for plugin '@brightside/db2':
___ Error: Unable to load the following module for plug-in '@brightside/db2' :
C:\Users\XXXXXXXX\.brightside\plugins\installed\node_modules\@brightside\db2\lib\imperative.js
Reason = The module '\\?\C:\Users\XXXXXXXX\.brightside\plugins\installed\node_modules\@brightside\db2\node_modules\ibm_db\build\Release\odbc_bindings.node'
was compiled against a different Node.js version using
NODE_MODULE_VERSION 59. This version of Node.js requires
NODE_MODULE_VERSION 64. Please try re-compiling or re-installing
the module (for instance, using `npm rebuild` or `npm install`).
___ Error: Failed to load the plugin's configuration from:
C:\Users\XXXXXXXX\.brightside\plugins\installed\node_modules\@brightside\db2\package.json
Reason = Cannot create property 'name' on string '{}'
No commands from this plugin will be available for future commands.
```

| App | Version |
|------|---------|
| Brightside | 1.0.1 |
| npm | 6.0.0 |
| Node | 10.6.0 |
| DB2 | 1.0.0 |

**Comment by AHumanFromCA** [comment]
Looks like we are waiting for this PR from the ibm_db folks.

ibmdb/node-ibm_db#398.

I would have expected that they would have been on top of the Node 10 LTS release today but seems like they haven't really cared about that.

**Comment by solsu01** [comment]
It seems that the ibmdb PR was merged for node 10 support. I just tried installing the db2 plugin on node 10 and it still fails:

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! ibm_db@2.3.0 install: `node installer/driverInstall.js`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the ibm_db@2.3.0 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/solsu01/.npm/_logs/2019-01-11T14_33_39_603Z-debug.log
Command Error:
Install Failed
